src/pages/[platform]/build-a-backend/data/working-with-files/index.mdx

```diff
@@ -1364,28 +1343,17 @@ class SongViewModel: ObservableObject {
         }
 
         // Get the song record
-        let result = try await Amplify.API.mutate(request: .get(Song.self, byId: currentSong.id))
-        guard case .success(let queriedSong) = result else {
-            print("Failed with error: ", result)
-            return
-        }
-        guard var song = queriedSong else {
-            print("Song may have been deleted, no song by id: ", currentSong.id)
-            await setCurrentSong(nil)
-            return
-        }
-        guard song.coverArtKey != nil else {
-            print("There is no cover art key to remove image association")
+        var song = try await Amplify.API.mutate(request: .get(Song.self, byIdentifier: currentSong.id)).get()
+
+        guard song?.coverArtPath != nil else {
+            print("There is no cover art path to remove image association")
             return
         }
 
         // Set the association to nil and update it
-        song.coverArtKey = nil
-        let updateResult = try await Amplify.API.mutate(request: .update(song))
-        guard case .success(let updatedSong) = updateResult else {
-            print("Failed with error: ", result)
-            return
-        }
+        song?.coverArtPath = nil
+
+        let updatedSong = try await Amplify.API.mutate(request: .update(song!)).get()
```

Comment:
guarding on `song` above will make this `song.coverArtPath = nil` and remove force cast for the update: `.update(song)`